infer_structcol/model.py

```diff
@@ -84,34 +88,46 @@ def calc_resid_spect(spect1, spect2):
                     sigma_r = sigma_eff_r, transmittance = residual_t, 
                     sigma_t = sigma_eff_t)
 
-def calc_log_prior(theta):
+def calc_log_prior(theta, theta_range):
     '''
     Calculates log of prior probability of obtaining theta.
     
     Parameters
     -------
-    theta: 3-, 5-tuple 
-        set of inference parameter values - volume fraction, baseline loss, 
-        wavelength dependent loss
+    theta: 5-, 7-tuple 
+        set of inference parameter values - volume fraction, particle radius, 
+        thickness, baseline loss, wavelength dependent loss
+    theta_range: 2 by 3 array of floats
+        user's best guess of the expected ranges of the parameter values 
+        ([[min_phi, max_phi], [min_radius, max_radius], [min_thickness, max_thickness]]) 
+    
     '''
-    if len(theta) == 5:
-        vol_frac, l0_r, l1_r, l0_t, l1_t = theta
+    if len(theta) == 7:
+        vol_frac, radius, thickness, l0_r, l1_r, l0_t, l1_t = theta
         if l0_r < 0 or l0_r > 1 or l0_r+l1_r <0 or l0_r+l1_r > 1:
             # Losses are not in range [0,1] for some wavelength
             return -np.inf 
         if l0_t < 0 or l0_t > 1 or l0_t+l1_t <0 or l0_t+l1_t > 1:
             # Losses are not in range [0,1] for some wavelength
             return -np.inf 
-    if len(theta) == 3:
-        vol_frac, l0, l1 = theta
+    if len(theta) == 5:
+        vol_frac, radius, thickness, l0, l1 = theta
         if l0 < 0 or l0 > 1 or l0+l1 <0 or l0+l1 > 1:
             # Losses are not in range [0,1] for some wavelength
             return -np.inf 
 
-    if not min_phi < vol_frac < max_phi:
+    if not theta_range[0,0] < vol_frac < theta_range[0,1]:
         # Outside range of validity of multiple scattering model
```

Comment:
In this comment, I wouldn't say that the value is necessarily outside of the range of validity of the model. It may just be outside the possible range given what we know about our systems.